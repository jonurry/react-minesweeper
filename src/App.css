.App {
  text-align: center; }

/* Define a variable at the top level of the cascade so that it is easily accessible to react */
:root {
  --primary-light: #9be7ff;
  --primary-main: #64b5f6;
  --primary-dark: #2286c3;
  --primary-contrast-text: #263238;
  --secondary-light: #ffe97d;
  --secondary-main: #ffb74d;
  --secondary-dark: #c88719;
  --secondary-contrast-text: #263238; }

.minefield {
  --columns: 8;
  --content-width: 96vw;
  --gutter: 1vw;
  --row-size: calc(
    (var(--content-width) - (var(--gutter) * (var(--columns) - 1))) /
      var(--columns)
  );
  display: grid;
  width: 100%;
  margin: calc(2vw + 64px) auto 2vw auto;
  max-width: var(--content-width);
  grid-template-columns: repeat(var(--columns), 1fr);
  grid-auto-rows: var(--row-size);
  grid-column-gap: var(--gutter);
  grid-row-gap: var(--gutter);
  justify-items: center;
  align-items: center;
  font-size: calc(70vw / var(--columns));
  line-height: var(--row-size); }
  .minefield > div svg {
    margin: auto;
    height: 100%; }

.scene {
  width: 100%;
  height: 100%;
  perspective: calc(var(--row-size) * 3); }

.card {
  width: 100%;
  height: 100%;
  position: relative;
  transition: transform 1s;
  transform-style: preserve-3d; }
  .card .front {
    background: var(--secondary-light);
    color: var(--primary-dark); }
  .card .back {
    color: var(--secondary-main);
    transform: rotateY(180deg); }
  .card.reveal {
    transform: rotateY(180deg); }
  .card.delay-0 {
    transition-delay: 0ms; }
  .card.delay-1 {
    transition-delay: 100ms; }
  .card.delay-2 {
    transition-delay: 200ms; }
  .card.delay-3 {
    transition-delay: 300ms; }
  .card.delay-4 {
    transition-delay: 400ms; }
  .card.delay-5 {
    transition-delay: 500ms; }

.card-face {
  backface-visibility: hidden;
  border-radius: 50%;
  height: 100%;
  position: absolute;
  width: 100%; }

.pulse {
  animation: pulsate 1s ease-in-out;
  animation-iteration-count: infinite; }

@keyframes pulsate {
  0% {
    width: 60%;
    height: 60%;
    margin: 20%;
    opacity: 0.6;
    stroke-width: 3; }
  50% {
    width: 80%;
    height: 80%;
    margin: 10%;
    opacity: 1;
    stroke-width: 5; }
  100% {
    width: 60%;
    height: 60%;
    margin: 20%;
    opacity: 0.6;
    stroke-width: 3; } }
