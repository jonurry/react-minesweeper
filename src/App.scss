.App {
  text-align: center;
}

/* Define a variable at the top level of the cascade so that it is easily accessible to react */
:root {
  // colour palette
  --primary-light: #9be7ff;
  --primary-main: #64b5f6;
  --primary-dark: #2286c3;
  --primary-contrast-text: #263238;
  --secondary-light: #ffe97d;
  --secondary-main: #ffb74d;
  --secondary-dark: #c88719;
  --secondary-contrast-text: #263238;
}

.minefield {
  // default to 8 columns. This will be changed by react on demand.
  --columns: 8;
  // The content width you use on your website
  --content-width: 96vw;
  // The size of the gutter
  --gutter: 1vw;
  // This is the calculation for the row height.
  --row-size: calc(
    (var(--content-width) - (var(--gutter) * (var(--columns) - 1))) /
      var(--columns)
  );

  display: grid;

  width: 100%;
  margin: 2vw auto;
  max-width: var(--content-width);

  grid-template-columns: repeat(var(--columns), 1fr);
  grid-auto-rows: var(--row-size);

  grid-column-gap: var(--gutter);
  grid-row-gap: var(--gutter);

  justify-items: center;
  align-items: center;

  font-size: calc(70vw / var(--columns));
  line-height: var(--row-size);

  > div {
    align-self: stretch;
    background: var(--secondary-light);
    border-radius: 50%;
    color: var(--secondary-contrast-text);
    justify-self: stretch;
    svg {
      margin: auto;
      height: 100%;
    }
  }
}
