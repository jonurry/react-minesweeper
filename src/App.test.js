import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

it('renders without crashing (ReactDOM)', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing (ReactRenderer)', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
