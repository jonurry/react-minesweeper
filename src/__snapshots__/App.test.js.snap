// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing (ReactRenderer) 1`] = `
<div
  className="App"
>
  <header
    className="MuiPaper-root-139 MuiPaper-elevation4-145 MuiAppBar-root-131 MuiAppBar-positionFixed-132 MuiAppBar-colorPrimary-137 mui-fixed"
  >
    <div
      className="MuiToolbar-root-166"
      style={
        Object {
          "margin": "auto",
        }
      }
    >
      <div
        className="MuiInput-root-173 MuiInput-underline-177 App-margin-129"
        color=""
      >
        <div
          className="MuiSelect-root-168"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-169 MuiSelect-selectMenu-170 MuiInput-input-181"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Easy
          </div>
          <input
            aria-invalid={undefined}
            aria-required={undefined}
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id="select-difficulty"
            name="difficulty"
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="hidden"
            value="easy"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-186 MuiSelect-icon-172"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <button
        aria-label="reset"
        className="MuiButtonBase-root-218 MuiButton-root-196 MuiButton-contained-205 MuiButton-fab-213 MuiButton-raised-208"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-197"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-186"
            color={undefined}
            focusable="false"
            style={
              Object {
                "color": "#2286c3",
                "fontSize": "48px",
              }
            }
            viewBox="0 0 24 24"
          >
            <g
              id="Bounding_Boxes"
            >
              <g
                id="ui_x5F_spec_x5F_header_copy_2"
              />
              <path
                d="M0,0h24v24H0V0z"
                fill="none"
              />
            </g>
            <g
              id="Duotone"
            >
              <g
                id="ui_x5F_spec_x5F_header_copy_3"
              />
              <g>
                <path
                  d="M12,4c-4.42,0-8,3.58-8,8c0,4.42,3.58,8,8,8s8-3.58,8-8C20,7.58,16.42,4,12,4z M15.5,8 C16.33,8,17,8.67,17,9.5S16.33,11,15.5,11S14,10.33,14,9.5S14.67,8,15.5,8z M8.5,8C9.33,8,10,8.67,10,9.5S9.33,11,8.5,11 S7,10.33,7,9.5S7.67,8,8.5,8z M12,17.5c-2.33,0-4.32-1.45-5.12-3.5h1.67c0.7,1.19,1.97,2,3.45,2s2.76-0.81,3.45-2h1.67 C16.32,16.05,14.33,17.5,12,17.5z"
                  opacity="0.3"
                />
                <circle
                  cx="15.5"
                  cy="9.5"
                  r="1.5"
                />
                <circle
                  cx="8.5"
                  cy="9.5"
                  r="1.5"
                />
                <path
                  d="M12,16c-1.48,0-2.75-0.81-3.45-2H6.88c0.8,2.05,2.79,3.5,5.12,3.5s4.32-1.45,5.12-3.5h-1.67C14.76,15.19,13.48,16,12,16z"
                />
                <path
                  d="M11.99,2C6.47,2,2,6.48,2,12c0,5.52,4.47,10,9.99,10C17.52,22,22,17.52,22,12C22,6.48,17.52,2,11.99,2z M12,20 c-4.42,0-8-3.58-8-8c0-4.42,3.58-8,8-8s8,3.58,8,8C20,16.42,16.42,20,12,20z"
                />
              </g>
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-221"
        />
      </button>
      <h3
        className="MuiTypography-root-228 MuiTypography-subheading-235 MuiTypography-colorInherit-247 App-margin-129"
      >
        <span
          className="MuiBadge-root-252 App-padding-130"
        >
          Mines
          <span
            className="MuiBadge-badge-253 MuiBadge-colorSecondary-255"
          >
            10
          </span>
        </span>
      </h3>
      <h1
        className="MuiTypography-root-228 MuiTypography-headline-233 MuiTypography-colorInherit-247 App-margin-129"
      >
        0:00
      </h1>
    </div>
  </header>
  <div
    className="minefield easy"
    style={
      Object {
        "--columns": 8,
        "--content-width": "calc(80vmin - 64px)",
        "--gutter": "1vh",
        "fontSize": "calc(var(--content-width) / var(--columns) * 0.6)",
      }
    }
  >
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
    <div
      className="scene"
    >
      <div
        className="card  delay-0"
        onClick={[Function]}
        onContextMenu={[Function]}
      >
        <div
          className="card-face front"
        >
          
        </div>
        <div
          className="card-face back"
        >
          
        </div>
      </div>
    </div>
  </div>
</div>
`;
